cmd/controller-manager/app/options/validation.go

```diff
@@ -24,11 +24,33 @@ func validateSecurePort(opts *Options) []error {
 	return errs
 }
 
+func validateDuration(opts *Options) []error {
```

Comment:
Generally looks good to me.
But please organize the validation by flags(not by type), thus, we can easily add more validations or remove them once they are deprecated.

Reply:
thanks for reminding~ I have updated the validation by flags